go/ql/lib/semmle/go/dataflow/internal/DataFlowUtil.qll

```diff
@@ -376,13 +376,8 @@ module BarrierGuard<guardChecksSig/3 guardChecks> {
         // Case: a function like "return someBarrierGuard(arg)"
         // or "return !someBarrierGuard(arg) && otherCond(...)"
         exists(boolean outcome |
-          ret = getUniqueOutputNode(fd, outp) and
           guardChecks(g, arg.asExpr(), outcome) and
-          // This predicate's contract is (p holds of ret ==> arg is checked),
-          // (and we have (this has outcome ==> arg is checked))
-          // but p.checkOn(ret, outcome, this) gives us (ret has outcome ==> p holds of this),
-          // so we need to swap outcome and (specifically boolean) p:
-          DataFlow::booleanProperty(outcome).checkOn(ret, p.asBoolean(), g)
+          guardingFunctionHelper(g, outp, p, fd, ret, outcome)
```

Comment:
Note the comment below ("See comment above") no longer has something to refer to